worker/src/main/scala/org/apache/celeborn/service/deploy/worker/Controller.scala

```diff
@@ -453,16 +453,11 @@ private[deploy] class Controller(
     }
 
     // Update shuffleMapperAttempts
-    shuffleMapperAttempts.putIfAbsent(shuffleKey, new AtomicIntegerArray(mapAttempts))
-    val attempts = shuffleMapperAttempts.get(shuffleKey)
-    if (mapAttempts.exists(_ != -1)) {
-      attempts.synchronized {
-        0 until attempts.length() foreach (idx => {
-          if (mapAttempts(idx) != -1 && attempts.get(idx) == -1) {
-            attempts.set(idx, mapAttempts(idx))
-          }
-        })
-      }
+    val shuffleMapAttempts = shuffleMapperAttempts.get(shuffleKey)
+    if (shuffleMapAttempts == null) {
+      shuffleMapperAttempts.putIfAbsent(shuffleKey, new AtomicIntegerArray(mapAttempts))
+    } else {
+      updateShuffleMapperAttempts(mapAttempts, shuffleMapAttempts)
```

Comment:
IMO, the updateShuffleMapperAttempts should be invoked even when shuffleMapAttempts is null.

Reply:
oh, right